docs/xrunjdwp.md

```diff
@@ -0,0 +1,122 @@
+
+### `server`
+
+        -Xrunjdwp:server=[y|n]
+
+: If `server=n`, the VM does not listen to any incoming debugger connection and it attempts to attach to the local debugger application at the specified [`address`](#address). In this case, specifying the`address` parameter is mandatory. If `server=y`, the target VM acts as a server listening for an incoming debugger connection through the JDWP agent. If `server=y` and `address` is specified, the VM listens for a debugger connection at this address. If `server=y` and `address` is not specified, the VM selects a [`transport`](#transport) address at which to listen for a debugger application.
+
+### `suspend`
+
+        -Xrunjdwp:suspend=[y|n]
+
+: If `suspend=y`, the VM suspends running of the target Java application until the debugger is attached successfully and is ready for debugging. If `suspend=n`, the VM does not suspend running of the target Java application while the debug connection is established. The VM continues running the application as normal. The debugger can attach to the running application at any time.
+
+: When a debug session is restored from a checkpoint, the VM suspends the running of the Java application that was being debugged even if `suspend=n`. The VM waits for the debugger to send a `Resume` request before it restarts the Java application to continue the debug session. When the debug session is complete, the debugger disconnects from the JDWP agent, and the agent resumes normal running of the Java application.
+
+### ![Start of content that applies to Java 11 (LTS) and later](cr/java11plus.png) `suspendOnRestore`
+
+**(Linux&reg; x86, Linux on POWER&reg; (Little Endian), Linux on AArch64, and Linux on IBM Z&reg; only)**
+
+`suspendOnRestore` is an OpenJ9 specific parameter and is similar to the `suspend` parameter. But unlike `suspend`, this parameter only affects a VM on restore.
+
+        -Xrunjdwp:suspendOnRestore=[y|n]
+
+: If `suspendOnRestore=y`, the VM suspends running of the target Java application until the debugger is attached successfully and is ready for debugging on CRIU restore. If `suspendOnRestore=n`, the VM does not suspend the running of the target Java application while the debug connection is established on restore.
+
+: If [`suspend=y`](#suspend) is set pre-checkpoint, the VM suspends the application pre-checkpoint, and on restore. The application remains suspended when you restore the VM even if you specify `suspend=n` pre-checkpoint. Therefore, to ensure that the application is not suspended on restore, you can use the `suspendOnRestore=n` setting.
+
+: The `suspendOnRestore` parameter is also useful if no debugging is set pre-checkpoint, but the debugging is enabled on restore.
+The debugger registers the post-restore event, receives notification upon CRIU restore, and suspends the target VM application unless `suspendOnRestore=n` is specified.
```

Comment:
If no debugging was set pre-checkpoint, the debugging can be enabled via a restore option file and `CRIUSupport.registerRestoreOptionsFile` API.
The three command-line options that worked pre-checkpoint can be specified in the restore option file, and all sub-options including `suspendOnRestore` work similarly.